src/modules/avformat/link_avdeinterlace.c

```diff
@@ -0,0 +1,617 @@
+			goto fail;
+		}
+		prev_ctx = avfilter_ctx;
+	} else {
+		const AVFilter* deint = (AVFilter*)avfilter_get_by_name( "bwdif" );
+		avfilter_ctx = avfilter_graph_alloc_filter( pdata->avfilter_graph, deint, deint->name );
+		if( !avfilter_ctx ) {
+			mlt_log_error( self, "Cannot create bwdif filter\n" );
+			goto fail;
+		}
+		ret = avfilter_init_str( avfilter_ctx, "mode=send_frame:parity=auto:deint=all" );
+		if( ret < 0 ) {
+			mlt_log_error( self, "Cannot init bwdif filter: %s\n", av_err2str(ret) );
+			goto fail;
+		}
+		ret = avfilter_link( prev_ctx, 0, avfilter_ctx, 0 );
+		if( ret < 0 ) {
+			mlt_log_error( self, "Cannot link bwdif filter\n" );
+			goto fail;
+		}
+		prev_ctx = avfilter_ctx;
+	}
+
+	// Initialize the buffer sink filter context
+	pdata->avbuffsink_ctx = avfilter_graph_alloc_filter( pdata->avfilter_graph, buffersink, "out");
+	if( !pdata->avbuffsink_ctx ) {
+		mlt_log_error( self, "Cannot create image buffer sink\n" );
+		goto fail;
+	}
+	ret = av_opt_set_int_list( pdata->avbuffsink_ctx, "pix_fmts", out_pixel_fmts, -1, AV_OPT_SEARCH_CHILDREN );
+	if( ret < 0 ) {
+		mlt_log_error( self, "Cannot set sink pixel formats %d\n", out_pixel_fmts[0] );
+		goto fail;
+	}
+	ret = avfilter_init_str(  pdata->avbuffsink_ctx, NULL );
+	if( ret < 0 ) {
+		mlt_log_error( self, "Cannot init buffer sink\n" );
+		goto fail;
+	}
+	ret = avfilter_link( prev_ctx, 0, pdata->avbuffsink_ctx, 0 );
+	if( ret < 0 ) {
+		mlt_log_error( self, "Cannot link filter to sink\n" );
+		goto fail;
+	}
+
+	// Configure the graph.
+	ret = avfilter_graph_config( pdata->avfilter_graph, NULL );
+	if( ret < 0 ) {
+		mlt_log_error( self, "Cannot configure the filter graph\n" );
+		goto fail;
+	}
+
+	return;
+
+fail:
+	avfilter_graph_free( &pdata->avfilter_graph );
+}
+
+static void link_configure( mlt_link self, mlt_profile chain_profile )
+{
+	private_data* pdata = (private_data*)self->child;
+
+	// Operate at the same frame rate as the next link
+	mlt_service_set_profile( MLT_LINK_SERVICE( self ), mlt_service_profile( MLT_PRODUCER_SERVICE( self->next ) ) );
+
+	// The source may have changed so restart sending future frames.
+	pdata->deinterlace_required = 1;
+	pdata->scan_mode_detected = 0;
+}
+
+static int link_get_image( mlt_frame frame, uint8_t **image, mlt_image_format *format, int *width, int *height, int writable )
+{
+	int error = 0;
+	int ret = 0;
+	mlt_link self = (mlt_link)mlt_frame_pop_service( frame );
+	private_data* pdata = (private_data*)self->child;
+	mlt_properties frame_properties = MLT_FRAME_PROPERTIES( frame );
+
+	struct mlt_image_s srcimg = {0};
+	struct mlt_image_s dstimg = {0};
+	mlt_deinterlacer method = mlt_deinterlacer_id( mlt_properties_get( frame_properties, "consumer.deinterlacer" ) );
+
+	if ( !mlt_properties_get_int( frame_properties, "consumer.progressive" ) ||
+		 method == mlt_deinterlacer_none )
+	{
+		mlt_log_info( MLT_LINK_SERVICE(self), "Do not deinterlace - consumer is interlaced\n" );
```

Comment:
Everywhere there is `mlt_log_info` needs reevaluation about that log level. Shotcut release builds log at info level; so, there could be many new messages flooding its log. Shotcut melt jobs and many melt users run with verbose level. So, some should be verbose and maybe some debug. Debug output is usually too much, and I often change certain log calls to verbose or info to investigate something and then change it back.

Reply:
I think we are mostly on the same page. I find it helpful when the Shotcut log includes some one-time informational messages. For example, when a resources is allocated for the first time in a filter. I do not use mlt_log_info() for recurring messages. I did change a couple of log levels in this last patch.